File: loader/src/main/java/de/timolia/lactea/loader/startup/LoadContext.java
package de.timolia.lactea.loader.startup;

import com.google.inject.Module;
import de.timolia.lactea.loader.inject.InjectedInstance;

/**
 * @author David (_Esel)
 */
public interface LoadContext {
    void installGlobalModule(Module module);

    void installModule(InjectedInstance<Module> module);
}

File: module-api/src/main/kotlin/de/timolia/lactea/lifecycle/LoadContext.kt
package de.timolia.lactea.lifecycle

import com.google.inject.Module;
import de.timolia.lactea.inject.InjectedInstance

interface LoadContext {
    fun installGlobalModule(module: InjectedInstance<Module>)
    fun installModule(module: InjectedInstance<Module>)

    var encapsulationMode: EncapsulationMode


    enum class EncapsulationMode {
        PRIVATE_MODULE,
        INJECTOR
    }
}
